### PR TITLE
[384][431] Servo 4in1, out of range int bug updated

### PR DESCRIPTION
**Pull request by kuba8686**

### CODE REVIEW

File: substations/06_ServoControllerHVAC/06_ServoControllerHVAC.ino [hunk 1]
```diff
@@ -4,16 +4,46 @@
...
+#define SERVO_1H_PIN 16 /* D0 */
+#define SERVO_1C_PIN 5  /* D1 */
+#define SERVO_2H_PIN 4  /* D2 */
+#define SERVO_2C_PIN 0  /* D3 */
+
+/* defining ID's for Servos */
+const int id_1H = Servo_1H_ID; /* 101 */
+const int id_1C = Servo_1C_ID; /* 102 */
+const int id_2H = Servo_2H_ID; /* 103 */
+const int id_2C = Servo_2C_ID; /* 104 */
```
Comment (szymoneiro) [inline substations/06_ServoControllerHVAC/06_ServoControllerHVAC.ino:17]: Are these variables values taken from init_config.h file? If so I think it's not neccecary to declare them once more, just use them as they are from init_config.h. Also then there should be added line: 
`#include "init_config.h`

Comment (kuba8686) [reply substations/06_ServoControllerHVAC/06_ServoControllerHVAC.ino:17]: Yes, they are taken from `init_config.h`, i forget to update init file at Wiki. The line `#include "init_config.h"` is already added at the line 1.

File: substations/06_ServoControllerHVAC/06_ServoControllerHVAC.ino [hunk 2]
```diff
@@ -27,34 +57,56 @@ String output_JSON(int a, int id, String t)
...
 
 void loop()
 {
-    network.loop();
     mqtt.loop();
-    mqtt.send(output_JSON(servoAng, id, type));
 
     if (++notifCaller == 1000) {
-        mqtt.send(output_JSON(servoAng, id, type));
+        mqtt.send(output_JSON(ang_1H, id_1H, type));
```
Comment (Siwy160) [inline substations/06_ServoControllerHVAC/06_ServoControllerHVAC.ino:106]: I don't like that naming. Hard to tell what does that mean.

File: substations/06_ServoControllerHVAC/06_ServoControllerHVAC.ino [hunk 3]
```diff
@@ -4,16 +4,46 @@
...
 int notifCaller = 0;
 
+int getPin_fromServoID(int id) 
+{
+    switch(id) {
+        case id_1H:  return SERVO_1H_PIN;
+        case id_1C:  return SERVO_1C_PIN;
+        case id_2H:  return SERVO_2H_PIN;
+        case id_2C:  return SERVO_2C_PIN;
+    }
```
Comment (cichy1024) [inline substations/06_ServoControllerHVAC/06_ServoControllerHVAC.ino:44]: It's OK but it should some kind of an array.

File: substations/06_ServoControllerHVAC/06_ServoControllerHVAC.ino [hunk 4]
```diff
@@ -27,34 +57,56 @@ String output_JSON(int a, int id, String t)
...
+    if(json_doc["type"].as<String>().equals(type) 
+       && json_doc["angle"] >= 0 
+       && json_doc["angle"] <=180) {
+        
+        switch(id) {
+            case id_1H: ang_1H = json_doc["angle"]; servo_1H.write(ang_1H); break;
+            case id_1C: ang_1C = json_doc["angle"]; servo_1C.write(ang_1C);break;
+            case id_2H: ang_2H = json_doc["angle"]; servo_2H.write(ang_2H);break;
+            case id_2C: ang_2C = json_doc["angle"]; servo_2C.write(ang_2C);break;
+        }
```
Comment (cichy1024) [inline substations/06_ServoControllerHVAC/06_ServoControllerHVAC.ino:80]: The same - I don't like those switches but let's leave it as is.

**Pull request merged**